src/v/cluster/partition.h

```diff
@@ -244,6 +244,21 @@ class partition {
         return _cloud_storage_partition->make_reader(config, deadline);
     }
 
+    ss::future<> remove_persistent_state() {
+        if (_rm_stm) {
+            co_await _rm_stm->remove_persistent_state();
+        }
+        if (_tm_stm) {
+            co_await _tm_stm->remove_persistent_state();
+        }
+        if (_archival_meta_stm) {
+            co_await _archival_meta_stm->remove_persistent_state();
+        }
+        if (_id_allocator_stm) {
+            co_await _id_allocator_stm->remove_persistent_state();
+        }
```

Comment:
does this mean that `id_allocator_stm` needs to be a regular ss::shared_ptr instead of the lightweight version? I don't think I have ever fully understood this nuance of the seastar shared pointers.

Reply:
good point, I've changed it - https://github.com/redpanda-data/redpanda/compare/b692ce2c6069d7a0d9572ef9aae9a8e2189e6ad3..a34277be012469a46b9eb8cb5fc3a1a85d072b62